x-pack/build_chromium/README.md

```diff
@@ -1,12 +1,115 @@
+We ship our own headless build of Chromium which is significantly smaller than the standard binaries shipped by Google. The scripts in this folder can be used to initialize the build environments and run the build on Mac, Windows, and Linux.
 
-# Building on macOS
-XCode and the OS X 10.10 SDK are currently required, but the full requirements are specified [here](https://chromium.googlesource.com/chromium/src/+/master/docs/mac_build_instructions.md). Also, you don't have to install the depot_tools.
+The official Chromium build process is poorly documented, and seems to have breaking changes fairly regularly. The build pre-requisites, and the build flags change over time, so it is likely that the scripts in this directory will be out of date by the time we have to do another Chromium build.
 
+This document is an attempt to note all of the gotchas we've come across while building, so that the next time we have to tinker here, we'll have a good starting point.
 
-## Updating the revision
-If you want to build the headless_shell that mirrors a specific version of the Chrome browser, you can take the version number displayed in `Chrome -> About` and plug it into the "Version" input [here](https://omahaproxy.appspot.com/) and it'll give you the Commit that you can feed into the build script using the `--git-sha` argument. It's highly recommended to update the default when you're upgrading the version.
+## Build args
+
+Chromium is built via a build tool called "ninja". The build can be configured by specifying build flags either in an "args.gn" file or via commandline args. We have an "args.gn" file per platform:
+
+- mac: darwin/args.gn
+- linux: linux/args.gn
+- windows: windows/args.gn
+
+The various build flags are not well documented. Some are documented [here](https://www.chromium.org/developers/gn-build-configuration). Some, such as `enable_basic_printing = false`, I only found by poking through 3rd party build scripts.
+
+As of this writing, there is an officially supported headless Chromium build args file for Linux: `build/args/headless.gn`. This does not work on Windows or Mac, so we have taken that as our starting point, and modified it until the Windows / Mac builds succeeded.
+
+## VMs
+
+I ran Linux and Windows VMs in GCP with the following specs:
+
+- 8 core vCPU
+- 30GB RAM
+- 128GB hard drive
+- Ubuntu 18.04 LTS (not minimal)
+- Windows Server 2016 (full, with desktop)
+
+The more cores the better, as the build makes effective use of each. For Linux, Ubuntu is the only officially supported build target.
+
+- Linux:
+  - SSH in using [gcloud](https://cloud.google.com/sdk/)
+  - Get the ssh command in the [GCP console](https://console.cloud.google.com/) -> VM instances -> your-vm-name -> SSH -> gcloud
```

Comment:
Should you add a step to start the vm instance if it is stopped?

Should we specifically call out the vm instance name so it matches whats on there now?  `chromium-build-linux` I see (but it's stopped so I can't SSH in)